src/test/regress/sql/multi_mx_master_update_node.sql

```diff
@@ -0,0 +1,52 @@
+-- Test creation of mx tables and metadata syncing
+
+SET citus.replication_model TO streaming;
+
+-- add a node to the cluster
+SELECT 1 FROM master_add_node('localhost', :worker_1_port);
+SELECT * FROM pg_dist_node;
+
+-- update the node
+SELECT 1 FROM master_update_node((SELECT nodeid FROM pg_dist_node),
+                                 'localhost', :worker_2_port);
+SELECT * FROM pg_dist_node;
+
+-- start syncing metadata to the node
+SELECT 1 FROM start_metadata_sync_to_node('localhost', :worker_2_port);
+SELECT * FROM pg_dist_node;
+
+-- Update the node again. We do this as epeatable read, so we just see the
+-- changes by master_update_node(). This is to avoid inconsistent results
+-- if the maintenance daemon does the metadata sync too fast.
+BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
+SELECT nodeid, nodename, nodeport, hasmetadata, metadatasynced FROM pg_dist_node;
+SELECT 1 FROM master_update_node((SELECT nodeid FROM pg_dist_node),
+                                 'localhost', :worker_1_port);
+SELECT nodeid, nodename, nodeport, hasmetadata, metadatasynced FROM pg_dist_node;
+END;
+
+-- wait until maintenance daemon does the next metadata sync, and then
+-- check if metadata is synced again
+SELECT pg_sleep(2);
```

Comment:
We shouldn't do this. Instead we should have a test UDF which waits for the next sync event.